pmd-core/src/main/java/net/sourceforge/pmd/RuleSet.java

```diff
@@ -48,6 +51,23 @@
     private Filter<File> filter;
 
     /**
+     * Creates a new RuleSet with the given checksum.
+     * @param checksum A checksum of the ruleset, should change
+     * only if the ruleset was configured differently
+     */
+    public RuleSet(final long checksum) {
+        this.checksum = checksum;
+    }
+    
+    /**
+     * Creates a new RuleSet with a random checksum.
+     * Notice this means the results for this analysis will not be cached.
+     */
+    public RuleSet() {
+        this(new Random().nextLong());
```

Comment:
Sounds good. The name reminds me a bit of Spring Framework's [Aware](http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/beans/factory/Aware.html) interface and subinterfaces, it probably works. I'll refactor it as soon as I get the chance. Yes, it's implemented as a `hashCode` on RuleSets. As a checksum I just want something that can be generated fast, and will change if anything changes.

Right now this second constructor is used only in unit tests, the designer, the benchmarker and XPathCLI. None of these should use the analysis cache, except maybe `XPathCLI`?

In future versions we may have `Rule` be `ChecksumAware` and have `RuleSet` compute their checksums off the `Rule`s, just like `RuleSets` does from `RuleSet` maybe?